### PR TITLE
Switch to bashbrew-based image builds

### PR DESCRIPTION
**Pull request by hairyhenderson**
This is in preparation to applying for an official build.

The build now uses [bashbrew](https://github.com/docker-library/official-images/tree/master/bashbrew), which is the canonical build tool for official images. Once we're ready to go for an official image, the `library/caddy` file will need to be PRed into https://github.com/docker-library/official-images/.

This also automatically regenerates `Dockerfile`s based on upstream image changes (from Renovate, for example). And the `library/caddy` file needs is regenerated automatically when `Dockerfile`s are updated (in separate commits, since the file references commit hashes)

We still need to do a bit more work in future PRs:

- the `Dockerfile`s need to be regenerated automatically based on new releases in https://github.com/caddyserver/caddy
- it'd also be good to consider building multi-arch images (supported by the config, we just need to make sure the builds actually work)
- the `README.md` needs updating!

This looks a bit more complicated than the previous one, since we can't give bashbrew a multi-stage target to build, and so it pretty much requires different image tags to be built from different Dockerfiles.

Signed-off-by: Dave Henderson <dhenderson@gmail.com>

### CODE REVIEW

File: Dockerfile.tmpl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+COPY --from=fetch-assets /Caddyfile /etc/caddy/Caddyfile
+COPY --from=fetch-assets /index.html /usr/share/caddy/index.html
+
+ARG VERSION=v2.0.0-beta.13
+LABEL org.opencontainers.image.version=$VERSION
+LABEL org.opencontainers.image.title=Caddy
+LABEL org.opencontainers.image.description="a powerful, enterprise-ready, open source web server with automatic HTTPS written in Go"
+LABEL org.opencontainers.image.url=https://caddyserver.com
+LABEL org.opencontainers.image.documentation=https://github.com/caddyserver/caddy/wiki/v2:-Documentation
```
Comment (Mohammed90) [inline Dockerfile.tmpl]:
```suggestion
LABEL org.opencontainers.image.documentation=https://caddyserver.com/docs
```

Comment (hairyhenderson) [reply Dockerfile.tmpl]: good call - I saw this and meant to update it, but forgot 😅

**Comment by hairyhenderson** [comment]
In chatting with @Ilyes512 we realized that this'll break auto-updates from renovate as-is. I'll need to think through that and figure out how to best address that.

**Comment by hairyhenderson** [comment]
Auto-updates with Renovate should work now! PTAL @Ilyes512 @Mohammed90 @mholt 🙂

**Pull request merged**